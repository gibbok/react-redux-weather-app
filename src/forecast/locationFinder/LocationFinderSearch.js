// @flow
import * as React from 'react'
import {Button, Input} from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +onLocationChange: Function,
  +onLocationKeyPress: Function,
  +onSearchClick: Function,
  +inputValue: string
}
/* eslint-enable no-undef */

const LocationFinderSearch:React.StatelessFunctionalComponent<any> = ({ onLocationChange, onLocationKeyPress, onSearchClick, inputValue }:PropsType):React.Element<any> => (
  <div>
    <Input onKeyPress={(e) => onLocationKeyPress(e)} placeholder='Enter a location' onChange={(e) => onLocationChange(e)} />
    <Button onClick={(e) => onSearchClick(e, inputValue)}>Search</Button>
  </div>
)

export default LocationFinderSearch
