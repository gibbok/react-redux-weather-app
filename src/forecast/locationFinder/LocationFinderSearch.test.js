import React from 'react'
import { mount, shallow } from 'enzyme'
import LocationFinderSearch from './LocationFinderSearch'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'

describe('<LocationFinderSearch />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <LocationFinderSearch
        onLocationChange={() => {}}
      />
      )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should mount in a full DOM', () => {
    expect(mount(<LocationFinderSearch />).find('form').length).toBe(1)
    expect(mount(<LocationFinderSearch />).find('input[type=\'text\']').length).toBe(1)
    expect(mount(<LocationFinderSearch />).find('input[type=\'submit\']').length).toBe(1)
  })

  it('should accept change', () => {
    // given
    const change = sinon.spy()
    const actualNode = shallow(
      <LocationFinderSearch
        onLocationChange={change}
    />)
    // when
    actualNode.find('input[type=\'text\']').simulate('change')
    // then
    sinon.assert.called(change)
  })

  it('should accept click on submit', () => {
    const onInputClick = sinon.spy()
    const actualNode = shallow(
      <LocationFinderSearch
        onSearchClick={onInputClick}
    />)
    actualNode.find('input[type=\'submit\']').simulate('click')
    sinon.assert.called(onInputClick)
  })
})
