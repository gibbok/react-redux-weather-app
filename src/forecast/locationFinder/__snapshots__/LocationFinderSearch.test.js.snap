// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationFinderSearch /> should render 1`] = `
<form>
  <input
    onChange={[Function]}
    placeholder="Location"
    type="text"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Search"
  />
</form>
`;
